presto-main/src/main/java/io/prestosql/server/security/InsecureAuthenticator.java

```diff
@@ -48,8 +60,14 @@ public Identity authenticate(ContainerRequestContext request)
             throw new AuthenticationException("Basic authentication or " + PRESTO_USER + " must be sent", BasicAuthCredentials.AUTHENTICATE_HEADER);
         }
 
-        return Identity.forUser(user)
-                .withPrincipal(new BasicPrincipal(user))
-                .build();
+        try {
+            String authenticatedUser = userMapping.mapUser(user);
+            return Identity.forUser(authenticatedUser)
+                    .withPrincipal(new BasicPrincipal(authenticatedUser))
```

Comment:
Shouldn't this be the original user? That's what we do for `PasswordAuthenticator` (at least for the built-in ones, as a custom password authenticator can return any principal).